apps/fishing-map/features/search/TrackFootprint.tsx

```diff
@@ -0,0 +1,132 @@
+import { useEffect, useRef, useState } from 'react'
+import cx from 'classnames'
+import { geoEqualEarth, geoPath } from 'd3'
+import { FeatureCollection, GeoJsonProperties, Geometry } from 'geojson'
+import { DateTime } from 'luxon'
+import { GFWAPI } from '@globalfishingwatch/api-client'
+import { Icon, Spinner } from '@globalfishingwatch/ui-components'
+import { segmentsToGeoJSON, trackValueArrayToSegments } from '@globalfishingwatch/data-transforms'
+import { Field } from '@globalfishingwatch/api-types'
+import { useOnScreen } from 'hooks/screen.hooks'
+import styles from './TrackFootprint.module.css'
+
+type TrackFootprintProps = {
+  vesselId: string
+  trackDatasetId?: string
+  highlightedYear?: number
+}
+
+const FOOTPRINT_WIDTH = 300
+const FOOTPRINT_HEIGHT = 150
+
+const PROJECTION = geoEqualEarth()
+  .scale(53.5)
+  .translate([FOOTPRINT_WIDTH / 2, FOOTPRINT_HEIGHT / 2])
+
+function TrackFootprint({ vesselId, trackDatasetId, highlightedYear }: TrackFootprintProps) {
+  const [trackData, setTrackData] = useState<FeatureCollection<Geometry, GeoJsonProperties>>()
+  const [error, setError] = useState(false)
+  const fullCanvasRef = useRef<HTMLCanvasElement | null>(null)
+  const highlightCanvasRef = useRef<HTMLCanvasElement | null>(null)
+  const onScreen = useOnScreen(fullCanvasRef)
+  const fullContext = fullCanvasRef.current?.getContext('2d')
+  const highlightContext = highlightCanvasRef.current?.getContext('2d')
+
+  const fetchData = async (vesselId: string) => {
+    if (!trackDatasetId) {
+      setError(true)
+      return
+    }
+    let data = await GFWAPI.fetch<any>(
```

Comment:
At some point, we should resolve this URL with the `resolveEndpoint` function, but this is enough for now to start testing